src/ros_filter.cpp

```diff
@@ -430,6 +448,119 @@ namespace RobotLocalization
+        worldBaseLinkTransMsg_.transform.rotation = filteredPosition.pose.pose.orientation;
+
+        // If the worldFrameId_ is the odomFrameId_ frame, then we can just send the transform. If the
+        // worldFrameId_ is the mapFrameId_ frame, we'll have some work to do.
+        if(filteredPosition.header.frame_id == odomFrameId_)
+        {
+          worldBaseLinkTransMsg_.header.stamp = filteredPosition.header.stamp + tfTimeOffset_;
+          worldBaseLinkTransMsg_.header.frame_id = filteredPosition.header.frame_id;
+          worldBaseLinkTransMsg_.child_frame_id = filteredPosition.child_frame_id;
+          worldTransformBroadcaster_.sendTransform(worldBaseLinkTransMsg_);
+        }
+        else if(filteredPosition.header.frame_id == mapFrameId_)
+        {
+          try
+          {
+            tf2::Transform worldBaseLinkTrans;
+            tf2::fromMsg(worldBaseLinkTransMsg_.transform, worldBaseLinkTrans);
+
+            tf2::fromMsg(tfBuffer_.lookupTransform(baseLinkFrameId_, odomFrameId_, ros::Time(0)).transform,
+                         odomBaseLinkTrans_);
+
+            /*
+             * First, see these two references:
+             * http://wiki.ros.org/tf/Overview/Using%20Published%20Transforms#lookupTransform
+             * http://wiki.ros.org/geometry/CoordinateFrameConventions#Transform_Direction
+             * We have a transform from mapFrameId_->baseLinkFrameId_, but it would actually transform
+             * a given pose from baseLinkFrameId_->mapFrameId_. We then used lookupTransform, whose
+             * first two arguments are target frame and source frame, to get a transform from
+             * baseLinkFrameId_->odomFrameId_. However, this transform would actually transform data
+             * from odomFrameId_->baseLinkFrameId_. Now imagine that we have a position in the
+             * mapFrameId_ frame. First, we multiply it by the inverse of the
+             * mapFrameId_->baseLinkFrameId, which will transform that data from mapFrameId_ to
+             * baseLinkFrameId_. Now we want to go from baseLinkFrameId_->odomFrameId_, but the
+             * transform we have takes data from odomFrameId_->baseLinkFrameId_, so we need its
+             * inverse as well. We have now transformed our data from mapFrameId_ to odomFrameId_.
+             * However, if we want other users to be able to do the same, we need to broadcast
+             * the inverse of that entire transform.
+            */
+
+            mapOdomTrans_.mult(worldBaseLinkTrans, odomBaseLinkTrans_);
+
+            mapOdomTransMsg_.transform = tf2::toMsg(mapOdomTrans_);
+            mapOdomTransMsg_.header.stamp = filteredPosition.header.stamp + tfTimeOffset_;
+            mapOdomTransMsg_.header.frame_id = mapFrameId_;
+            mapOdomTransMsg_.child_frame_id = odomFrameId_;
+
+            worldTransformBroadcaster_.sendTransform(mapOdomTransMsg_);
+          }
+          catch(...)
```

Comment:
I guess this catches the tf exceptions. I'd be better if you catch the actual exceptions rather than all with `...`; I think that will tell more detailed reasons about why the exception was triggered.